### PR TITLE
Rename notification endpoint and add option to send email

### PR DESCRIPTION
**Pull request by kburke**
Initial commit for ingest-api issue #328.  Rename the notification endpoint to /notify, and add an option to email a message once it posts to Slack.

### CODE REVIEW

**Comment by kburke** [comment]
BTW @yuanzhou Flask Mail 0.9.1 from 2014 continues to be the most current I find in the changelog!  It looks to be an extension, barely mentioned at palletsprojects.com.
https://pythonhosted.org/Flask-Mail/changelog.html
https://pypi.org/project/Flask-Mail/
https://github.com/mattupstate/flask-mail

**Comment by kburke** [comment]
Per Bill's guidance at 6/23/2023 Standup, we'll configure Dev with `data-testing-notifications` as the channel and probably `kburke@pitt.edu` as the "admin email" for verification of the `/notify` endpoint.  I suggest we leave this configuration on Dev until Jes finishes integration with `ingest-ui`.

When this code gets to Test, we can configure with the channel and email address used by data curation/ingest staff, and coordinate a test with them.

**Comment by yuanzhou** [comment]
Thanks @kburke for the notes. Can you please also branch off the `main` in https://github.com/hubmapconsortium/gateway and send a PR to `dev-integrate` after renaming the endpoint from `notify-slack` to `/notify`? Will just need to rename in those `api_endpoints.<dev|test|stage|prod>.json` files. This will complete the endpoint renaming and I'll redeploy both.

**Comment by kburke** [comment]
Requested PR to dev-integrate in the gateway repo:
https://github.com/hubmapconsortium/gateway/pull/262

File: src/app.py [hunk 1]
```diff
@@ -17,8 +18,9 @@
 from requests.packages.urllib3.exceptions import InsecureRequestWarning
 import argparse
 from pathlib import Path
-from flask import Flask, g, jsonify, abort, request, json, Response
+from flask import Flask, g, jsonify, abort, request, json, Response, render_template
```
Comment (yuanzhou) [inline src/app.py:21]: @kburke this `render_template` is not being used

**Review by yuanzhou — CHANGES_REQUESTED** [review]

File: src/app.py (hunk 1 above)
Comment (kburke) [reply src/app.py]: Pushed change to remove, thanks!

**Review by yuanzhou — APPROVED** [review]

**Pull request merged**